### ISSUE TITLE
Experiencing error code 137 with EC2 runners

### ISSUE THREAD

**Issue by cmbailey87** [npalm/terraform-aws-gitlab-runner#346]
So I ran some pipelines from gitlab and everything ran successfully. 
I redeployed my ec2 gitlab runners, and am now experiencing error code 137 from the docker machine runners. 

I had tried increase the size of the ec2 instance  ( t3 small, medium , and large ) and tried increasing the storage 16/32/40 and even 100GB... but nothing seems to be getting by this issue. 

has there been an update that may be causing the issue with the docker machine runner resources?

**Comment by cmbailey87** [comment]
Looks like I figured out the issue..
In the latest version 4.26.0, the runner is being configured and upgrade to a gp3

This seems to be breaking my pipelines. I pinned my versioning to 4.25 since this version defaults to gp2. 
is there an option in the config to change runners to gp2 instead of gp3.. 
![image](https://user-images.githubusercontent.com/26728086/125833569-54399884-c357-4b83-9eb6-fca2ac5a7b0f.png)

**Comment by kayman-mk** [comment]
> Docker exit code 137 implies Docker doesn't have enough RAM to finish the work.
Did you check this? May be the docker version was also increased and needs more memory now.

**Comment by cmbailey87** [comment]
> > Docker exit code 137 implies Docker doesn't have enough RAM to finish the work.
> 
> Did you check this? May be the docker version was also increased and needs more memory now.

@kayman-mk 
I considered that an option aswell, but I am not sure how to directly increase the docker machines ram. 

What I did was increase my EC2 instances type.... which , by Doing that, automatically increases the available ram for the ec2.  the Ec2 itself that i tried were ( t3 small, medium , and large )  2/4/8/and 16 GB and ram on the EC2 and they all failed with the same docker error code 137.

**Comment by npalm** [comment]
I will check, still having the issue?

**Comment by cmbailey87** [comment]
> I will check, still having the issue?

I version pinned the gitlab runner module to version 4.25.0, the previous version. 
So far that is getting me by just fine. 

I would like be to able to use the latest version... is there an option to set the runner EC2 root image to a gp2 instead of gp3 during the runner deployment? So that I can test and filter that out as the possible issue with version 4.26.0

**Comment by npalm** [comment]
Yes you should set:

```hcl
runner_root_block_device = {
  delete_on_termination = true
  volume_type = gp2
  volume_size = 9
  encrypted = true
  iops = null
  kms_key_id = null
}

**Comment by cmbailey87** [comment]
requesting to leave this issue open a bit longer, I will be testing this out in my environment in a few weeks.
many thanks

**Comment by npalm** [comment]
@cmbailey87 any update?

**Comment by cmbailey87** [comment]
@npalm 
I used the latest version last week in a test, and received no errors regarding docker 137 resource errors
and didn't have to change the volume type to GP2 
I left everything as default (GP3 provisioned) and it worked. 

not sure what changed form version 4.26 ( where I got errors) to the latest 4.35, but it works now.
Thank you for your efforts and hard work on this projects. 

I'll close this issue out.
